study/cyber-tutor-2/js/app.js

```diff
@@ -0,0 +1,204 @@
+    correctAnswer: "CVEs",
+	responseVal: false
+  },
+];
+
+/*const ALL_PROBLEM_TYPES = [
+    "forLoop",
+    "nestedForLoop",
+    "whileLoop"
+];*/
+
+// globals
+let score = 0;
+//let selectedProblemTypes = JSON.parse(JSON.stringify(ALL_PROBLEM_TYPES));
+let problem, answer;
+let number = 0;
+// initialize UI components
+
+showProblem();
+
+setAnswer();
+//createProblemTypeCheckboxes();
+wordBank();
+// bind onclick functions to the buttons
+document.getElementById("submit").onclick = checkAnswer;
+
+function showProblem(){
+	problem = myQuestions[number].question;
+ // show problem text to user
+    updatePrompt(problem);
+
+}
+function setAnswer(){
+	answer = myQuestions[number].correctAnswer;
+}
+
+
+/*function createProblemTypeCheckboxes() {
+    // create checkboxes for the user to be able to customize problem types that they practice
+    ALL_PROBLEM_TYPES.forEach(problemType => {
+        // create list item element
+        const li = document.createElement("li");
+
+        // create a checkbox element
+        const input = document.createElement("input");
+        input.type = "checkbox";
+        input.id = problemType;
+        input.name = problemType;
+        input.value = problemType;
+        input.checked = true;
+
+        // define click action to add/remove problem type from selected problem types
+        input.onclick = () => {
+            // get the value of the input element
+            const val = input.value;
+
+            // if the input is checked, add to list of selected problem types
+            if (input.checked) {
+                selectedProblemTypes.push(val);
+            }
+            // if input is unchecked, remove it from the list of selected problem types
+            else {
+              //  selectedProblemTypes.splice(selectedProblemTypes.indexOf(val), 1);
+            }
+
+            // update prompt
+            promptText = showProblem();//generateProblem();
+            updatePrompt(promptText);
+        }
+
+        // create label for checkbox input
+        const label = document.createElement("label");
+       // label.htmlFor = problemType;
+       // label.innerText = problemType;
+
+        // add checkbox and label to list item
+        li.appendChild(input);
+        li.appendChild(label);
+
+        // add list item to the problem types list element
+        //document.getElementById("problemTypes").appendChild(li);
+    });
+}
+*/
+
+
+function updatePrompt(code) {
+    const codeEl = document.getElementById("code");
+
+    // clear out anything that could still be in the code prompt element
+    codeEl.innerHTML = "";
+
+    // add code mirror className so syntax highlighting works
+    codeEl.className = "cm-s-default";
+
+    // run CodeMirror syntax highlighting on the code
+    CodeMirror.runMode(code, { name: "text/x-java" }, codeEl);
+}
+
+
+function wordBank(){
+	let word = "";
+	for(let i=0;i<myQuestions.length;i++){
```

Comment:
This could be done without having to use a for loop.
```js
let words = myQuestions.map(question => question.correctAnswer);
    
document.getElementById("wordbank").innerText = words.join(" * ");
```